### PR TITLE
Adding shared scope usage REST API

### PR DESCRIPTION
**Pull request by mushthaq33**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 1]
```diff
@@ -14704,6 +14706,80 @@ public String getSharedScopeKeyByUUID(String uuid) throws APIManagementException
...
+        }
+
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            String sqlQueryForApiUsage = SQLConstants.GET_SHARED_SCOPE_API_USAGE_BY_TENANT;
+            psForApiUsage = conn.prepareStatement(sqlQueryForApiUsage);
+            psForApiUsage.setString(1, uuid);
+            psForApiUsage.setInt(2, tenantId);
+            apiUsageResultSet = psForApiUsage.executeQuery();
+            while (apiUsageResultSet.next()) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:14743]: is it possible to use try with resources

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 2]
```diff
@@ -8825,6 +8861,74 @@ definitions:
...
+  #-----------------------------------------------------
+  # The API resource object where the shared scope is used
+  #-----------------------------------------------------
+  SharedScopeUsedAPIResourceInfo:
+    title: API resource object using shared scope
+    properties:
+      target:
+        type: string
+      verb:
+        type: string
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:8930]: shall we add examples here too?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 1 above)
Comment (mushthaq33) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Fixed with 3550c6ac829d5e756dab682eab507f16e5616d0b

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 2 above)
Comment (mushthaq33) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed with 3550c6ac829d5e756dab682eab507f16e5616d0b

**Pull request merged**